baker/algolia/utils/mdimViews.ts

```diff
@@ -0,0 +1,159 @@
+import {
+    ChartConfigsTableName,
+    DbEnrichedMultiDimDataPage,
+    DbPlainMultiDimXChartConfig,
+    DbRawChartConfig,
+    getUniqueNamesFromParentTagArrays,
+    merge,
+    multiDimDimensionsToViewId,
+    MultiDimXChartConfigsTableName,
+    parseChartConfig,
+} from "@ourworldindata/utils"
+import * as db from "../../../db/db.js"
+import { getAllPublishedMultiDimDataPages } from "../../../db/model/MultiDimDataPage.js"
+import { getAnalyticsPageviewsByUrlObj } from "../../../db/model/Pageview.js"
+import { logErrorAndMaybeCaptureInSentry } from "../../../serverUtils/errorLog.js"
+import { ChartRecordType } from "../../../site/search/searchTypes.js"
+import {
+    getRelevantVariableIds,
+    getRelevantVariableMetadata,
+} from "../../MultiDimBaker.js"
+
+async function getChartConfigsByIds(
+    knex: db.KnexReadonlyTransaction,
+    ids: string[]
+) {
+    const rows = await knex<DbRawChartConfig>(ChartConfigsTableName)
+        .select("id", "full")
+        .whereIn("id", ids)
+    return new Map(rows.map((row) => [row.id, parseChartConfig(row.full)]))
+}
+
+async function getMultiDimXChartConfigIdMap(trx: db.KnexReadonlyTransaction) {
+    const rows = await trx<DbPlainMultiDimXChartConfig>(
+        MultiDimXChartConfigsTableName
+    ).select("id", "multiDimId", "viewId")
+    return new Map(
+        rows.map((row) => [`${row.multiDimId}-${row.viewId}`, row.id])
+    )
+}
+
+async function getRecords(
+    trx: db.KnexReadonlyTransaction,
+    multiDim: DbEnrichedMultiDimDataPage,
+    tags: string[],
+    pageviews: Record<string, { views_7d: number }>
+) {
+    const { slug } = multiDim
+    console.log(
+        `Creating ${multiDim.config.views.length} records for mdim ${slug}`
+    )
+    const multiDimXChartConfigIdMap = await getMultiDimXChartConfigIdMap(trx)
+    const chartConfigs = await getChartConfigsByIds(
+        trx,
+        multiDim.config.views.map((view) => view.fullConfigId)
+    )
+    const relevantVariableIds = getRelevantVariableIds(multiDim.config)
+    const relevantVariableMetadata =
+        await getRelevantVariableMetadata(relevantVariableIds)
+    return multiDim.config.views.map((view) => {
+        const viewId = multiDimDimensionsToViewId(view.dimensions)
+        const id = multiDimXChartConfigIdMap.get(`${multiDim.id}-${viewId}`)
+        if (!id) {
+            throw new Error(
+                `MultiDimXChartConfig not found multiDimId=${multiDim.id} viewId=${viewId}`
+            )
+        }
+        const chartConfig = chartConfigs.get(view.fullConfigId)
+        if (!chartConfig) {
+            throw new Error(
+                `MultiDim view chart config not found id=${multiDim.id} ` +
+                    `viewId=${viewId} chartConfigId=${view.fullConfigId}`
+            )
+        }
+        const params = new URLSearchParams(view.dimensions)
+        const queryStr = `?${params.toString()}`
```

Comment:
nit: use `queryParamsToStr` here (from utils)
```suggestion
        const queryStr = queryParamsToStr(view.dimensions)
```

Reply:
Thanks, changed.